electricitylci/petroleum_upstream.py

```diff
@@ -0,0 +1,118 @@
+
+def generate_petroleum_upstream(year):
+    """
+    Generate the annual petroleum extraction, transport, and refining emissions
+    (in kg) for each plant in EIA923.
+    
+    Parameters
+    ----------
+    year: int
+        Year of EIA-923 fuel data to use.
+    
+    Returns
+    ----------
+    dataframe
+    """
+    eia_fuel_receipts_df=read_eia923_fuel_receipts(year)
+    petroleum_criteria = eia_fuel_receipts_df['fuel_group']=='Petroleum'
+    eia_fuel_receipts_df=eia_fuel_receipts_df.loc[petroleum_criteria,:]
+    
+    eia_fuel_receipts_df['heat_input']=(
+            eia_fuel_receipts_df['quantity']*
+            eia_fuel_receipts_df['average_heat_content']*
+            pq.convert(10**6,'Btu','MJ'))
+    
+    #Sum all fuel use by plant, plant state, and DFO/RFO
+    eia_fuel_receipts_df=eia_fuel_receipts_df.groupby(
+            ['plant_id','plant_state','energy_source'],
+            as_index=False)['heat_input'].sum()
+    
+    #Assume that plants have fuel delivered from the PADD they are in. Note
+    #that the crude represented in each PADD is the mix of crude going into
+    #that PADD, domestically-produced and imported, and the refining emissions
+    #are representative of the mix of refinery types in that PADD.
+    state_padd_df=pd.read_csv(data_dir+'/state_padd.csv')
+    state_padd_dict=pd.Series(
+            state_padd_df.padd.values,index=state_padd_df.state).to_dict()
+    
+    #Assign each power plant to a PADD.
+    eia_fuel_receipts_df['padd']=(
+            eia_fuel_receipts_df['plant_state'].map(state_padd_dict))
+    
+    #Creating a dictionary to store the inventories of each fuel/PADD
+    #combination
+    petroleum_lci={}
+    fuels=['Diesel','Bunker']
+    fuels_map={'Diesel':'DFO',
+               'Bunker':'RFO'
+            }
+    padds=[1,2,3,4,5]
+    expected_lci_folder = 'petroleum_inventory'
+    for fuel in fuels:
+        for padd in padds:
+            fn = f'PRELIM_Mixer__{fuel}___PADD_{padd}_.xlsx'
+            path = join(data_dir,expected_lci_folder,fn)
+            key = f'{fuels_map[fuel]}_{padd}'
+            petroleum_lci[key]=pd.read_excel(
+                    path,
+                    sheet_name='Inventory',
+                    usecols="I:N",
+                    skiprows=2)
+            petroleum_lci[key]['fuel_code']=f'{fuels_map[fuel]}_{padd}'
+    
+    #Merging the dataframes within the dictionary to a single datframe
+    combined_lci=pd.concat(petroleum_lci,ignore_index=True)
+    eia_fuel_receipts_df['fuel_padd']=(eia_fuel_receipts_df['energy_source']+
+                        '_'+eia_fuel_receipts_df['padd'].astype(str))
+    
+    #Merge the inventories for each fuel with the fuel use by each power plant
+    merged_inventory = combined_lci.merge(
+            right=eia_fuel_receipts_df[['plant_id','heat_input','fuel_padd']],
+            left_on='fuel_code',
+            right_on='fuel_padd',
+            how='left').sort_values(['plant_id','fuel_padd','Flow.1'])
```

Comment:
I'm getting an error here when running combinator.py. Some sort of keyerror for the column name Flow.1.
```
Traceback (most recent call last):
  File "combinator.py", line 379, in <module>
    petro_df = petro.generate_petroleum_upstream(2016)
  File "/Users/greg/Documents/NETL work/ElectricityLCI/electricitylci/petroleum_upstream.py", line 79, in generate_petroleum_upstream
    how='left').sort_values(['plant_id','fuel_padd','Flow.1'])
  File "/Users/greg/miniconda3/envs/elci/lib/python3.7/site-packages/pandas/core/frame.py", line 4411, in sort_values
    stacklevel=stacklevel)
  File "/Users/greg/miniconda3/envs/elci/lib/python3.7/site-packages/pandas/core/generic.py", line 1382, in _get_label_or_level_values
    raise KeyError(key)
KeyError: 'Flow.1'
```

Reply:
This caused by a more recent version of pandas. When reading the Excel files with duplicate column names, pandas will append the .0, .1 etc. to the end of the column name to avoid that duplication - it does this even if you specify a range that doesn't include the duplicate column names. 

Just pushed a fix - added column rename to get rid of the suffixes.